code/game/objects/effects/effect_system.dm

```diff
@@ -978,6 +981,75 @@ steam.start() -- spawns the effect
 	update_nearby_tiles()
 	..()
 
+/turf/simulated/floor/foamedmetal
+	name = "foamed metal floor"
+	desc = "A lightweight foamed metal floor."
+	icon_state = "foamedmetal"
+	icon_regular_floor = "foamedmetal"
+	icon_plating = "foamedmetal"
+	can_exist_under_lattice = 1
+
+/turf/simulated/floor/foamedmetal/attack_hand(mob/user as mob)
+	user.delayNextAttack(10)
+	if ((M_HULK in user.mutations) || (prob(50)))
+		to_chat(user, "<span class='notice'>You smash through \the [src].</span>")
+		for(var/mob/O in oviewers(user))
+			if ((O.client && !( O.blinded )))
+				to_chat(O, "<span class='warning'>[user] smashes through \the [src].</span>")
+		src.ChangeTurf(get_base_turf(src.z))
+	else
+		to_chat(user, "<span class='notice'>You hit \the [src] but bounce off it.</span>")
+	return
+
+/turf/simulated/floor/foamedmetal/attackby(obj/item/C as obj, mob/user as mob)
+	if(!(locate(/obj/structure/lattice) in contents))
+		if(istype(C, /obj/item/stack/rods))
+			return
+	else if(istype(C, /obj/item/stack/tile))
+		return
+	user.delayNextAttack(10)
+	if (istype(C, /obj/item/weapon/grab))
+		var/obj/item/weapon/grab/G = C
+		G.affecting.loc = src.loc
+		for(var/mob/O in viewers(src))
+			if (O.client)
+				to_chat(O, "<span class='warning'>[G.assailant] smashes [G.affecting] through \the [src].</span>")
+		returnToPool(C)
+		src.ChangeTurf(get_base_turf(src.z))
+		return
```

Comment:
Is it? If it gets here then the foam has been destroyed, it needs to not continue to the next if statement.